src/Common/src/System/SR.Core.cs

```diff
@@ -0,0 +1,124 @@
+        //
+
+        private static Object s_resourceMap;
+        private const string MoreInfoLink = @". For more information, visit http://go.microsoft.com/fwlink/?LinkId=623485";
+
+        private static Object ResourceMap
+        {
+            get
+            {
+                if (SR.s_resourceMap == null)
+                {
+                    SR.s_resourceMap = WinRTInterop.Callbacks.GetResourceMap(SR.s_resourcesName);
+                }
+                return SR.s_resourceMap;
+            }
+        }
+
+        // This method is used to decide if we need to append the exception message parameters to the message when calling SR.Format. 
+        // by default it returns false. We overwrite the implementation of this method to return true through IL transformer 
+        // when compiling ProjectN app as retail build. 
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static bool UsingResourceKeys()
+        {
+            return false;
+        }
+
+        // TODO: Resouce generation tool should be modified to call this version in release build
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        internal static string GetResourceString(string resourceKey)
+        {
+            return WinRTInterop.Callbacks.GetResourceString(ResourceMap, resourceKey, null);
+        }
+
+        internal static string GetResourceString(string resourceKey, string defaultString)
+        {
+            string resourceString = GetResourceString(resourceKey);
+
+            // if we are running on desktop, GetResourceString will just return resourceKey. so
+            // in this case we'll return defaultString (if it is not null). 
+            if (defaultString != null && resourceKey.Equals(resourceString))
+            {
+                return defaultString;
+            }
+
+            if (resourceString == null)
+            {
+                // It is not expected to have resourceString is null at this point.
+                // this means our framework resources is missing while it is expected to be there. 
+                // we have to throw on that or otherwise we�ll eventually get stack overflow exception.
+                // we have to use hardcode the exception message here as we cannot lookup the resources for other keys.
+                // We cannot throw MissingManifestResourceException as we cannot depend on the System.Resources here.
+
+                throw new InvalidProgramException("Unable to find resource for the key " + resourceKey + ".");
+            }
+
+            return resourceString;
+        }
+
+        internal static string Format(string resourceFormat, params object[] args)
+        {
+            if (args != null)
+            {
+                if (UsingResourceKeys())
+                {
+                    return resourceFormat + String.Join(", ", args) + MoreInfoLink;
```

Comment:
Nit: `String.Join` -> `string.Join` and `String.Format` -> `string.Format` throughout.